### PR TITLE
Remove EMA alert if EMA is complete for current period.

### PR DESCRIPTION
**Pull request by eschlange**
* Updated view to use new controller logic.
* Updated ema service to pull latest complete ema.
* Updated controller logic which determines if the ema alert is visible.
* Added specs for existing and new methods in home controller and ema service.

[#110667116]

### CODE REVIEW

File: src/js/controllers/HomeController.js [hunk 1]
```diff
@@ -52,16 +54,32 @@
...
         var notificationTime = moment(this.lastEmaNotification.at);
         var currentTime = moment();
         timeRemainingInMinutes = notificationTime.diff(currentTime, 'minutes') +
-          TIME_LIMIT_IN_MINUTES;
+          this.TIME_LIMIT_IN_MINUTES;
       }
       return timeRemainingInMinutes;
     };
+
+    this.showEmaAlert = function() {
```
Comment (michaelwehrley) [inline src/js/controllers/HomeController.js]: I got the logic; however, in the future it maybe easier to understand what is going on here if you did something like this:
```javascript
if (noEmas()) {
  isVisible = true;
} else {
  if (didntAnswerLastEma()) {
    isVisible = true;
  }
}
```

File: src/js/services/Ema.js [hunk 2]
```diff
@@ -18,6 +18,15 @@
     this.saveCigaretteLog = function(emaQuestions) {
       this.saveEma(emaQuestions, CIGARETTE_LOG_TYPE);
     };
+
+    this.getLastEma = function() {
+      var emaData = emaCache.fetchAllRaw();
+      var lastEma;
```
Comment (michaelwehrley) [inline src/js/services/Ema.js]: Did you want to set this to `null` since it may never get defined?

**Comment by michaelwehrley** [comment]
:+1: looks great!

**Comment by michaelwehrley** [comment]
:merge when ready! :ship:

File: src/js/services/Ema.js (hunk 2 above)
Comment (eschlange) [inline src/js/services/Ema.js]: Good idea, less ambiguity. I'll make the update to the service and the method's spec.

File: src/js/controllers/HomeController.js (hunk 1 above)
Comment (eschlange) [inline src/js/controllers/HomeController.js]: Understood, thanks!

**Pull request merged**